include/openssl/kdf.h

```diff
@@ -90,6 +93,19 @@ extern "C" {
             EVP_PKEY_CTX_ctrl_uint64(pctx, -1, EVP_PKEY_OP_DERIVE, \
                             EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES, maxmem_bytes)
 
+# define EVP_PKEY_CTX_set_pbkdf2_md(pctx, md) \
+            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
+                              EVP_PKEY_CTRL_PBKDF2_MD, 0, (void *)(md))
+
+# define EVP_PKEY_CTX_set1_pbkdf2_salt(pctx, salt, saltlen) \
```

Comment:
To me it depends on the meaning you attaching to the ctrl. The existing EVP_PKEY_CTRL_MD is used for signatures and indicates the digest to sign/verify with. In fact the code in EVP_PKEY_CTX_ctrl_str with the "digest" string only works for signatures.

If we want to generalise this to mean "the digest associated with the operation" which isn't just signatures then that at least will need changing.